### DISCUSSION TITLE
How can I access my vault when server is down?

### DISCUSSION THREAD

**Discussion by Iceman248** [dani-garcia/vaultwarden#2683] [Q&A] [answered]
I am having an issue that is much more significantly difficult to deal with when I cannot access my vault at all. My server is down, but every client I have immediately gives session expired when it cannot contact the server, which makes the data syncing to my client apps quite pointless when I cannot access the data without logging into the server. Is there a way to mitigate this? This would otherwise force me to keep important passwords outside of my vault simply because I can't count on accessing my vault from clients if server is ever down.

**Comment by Masgalor** [comment, marked as answer]
Just to be sure.
You are not asking for help with your server that has an unexpected downtime right now, but you are asking for the offline functionality of some client you have in use, right?

As for the clients, you may get better answers from the official bitwarden help channel.
Thats because all clients are developed by bitwarden itself and not by vaultwarden.

Anyway, I can access all passwords in the android app when my phone is offline e.g. put in airplane-mode.

Some helpfull links:
https://bitwarden.com/help/
https://bitwarden.com/help/using-bitwarden-offline/

**Reply by Iceman248** [reply]
Thanks. That is correct, although I figured I would start here as I didn't know how if using bitwarden's cloud would work comparatively to the server being self hosted. I was able to access from my phone one time, but then it always locks me out. You mentioning airplane mode. Are you saying that if I had this issue I should cut off my internet access and therefore it wouldn't sense the server not being available and therefore lock me out?

**Reply by Masgalor** [reply]
You got me wrong.
What I tried to say is, that if I shut my server down or disconnect my phone from the internet I am still able to use the cached data on my phone. I dont see a reason why you would not be able to unlock the database on your phone if your server is unavailable.

Maybe you get locked out if your server crashes during a full resync, because the database was corrupted that way.
This never happend to me or anyone I know, so I am not sure if it is a possible scenario.

**Reply by Iceman248** [reply]
There has been many times from different devices that my server has been inaccessible for different reasons and every time it always locks me out. The only exception is that I might one time be able to log in to the cached data before it locks me out, but that's just sometimes. It's as if it's designed to do that once it realizes the server is not available. I haven't seen any setting or any way to not make it do that.

I'll have to inquire some more to see if there's some way I can prevent or maybe I should make an additional vault of some kind at least on one of my devices that I can import into occasionally just in case it ever happens again.

**Reply by BlackDex** [reply]
Well, it could also be client settings.
If you configure you clients to `Log out` instead of `Lock`, then you always need to login.
If you only configure the clients to `Lock` the vault, then you should be able to use it offline of a long period of time.

Also, if you have configure 2FA/MFA and didn't check `Remember Device`, then you need to always provide this during login.

Also, there is nothing we can do on the server side regarding this.

**Reply by Iceman248** [reply]
I do use Lock and Remember Device. Thanks. I'll ask bitwarden directly.

**Reply by IIIdefconIII** [reply]
HI, i was questening myself this today, how about mfa, did you a a permanent easy fix? if i putt over 30 2fa keys to my vaultwarden, what happens when the server goes blacked out and even to recieve a email or text as failover isnt configured?

**Comment by vmvmpvm** [comment]
I have seen this behavior when the reverse proxy is available but not the actual vaultwarden instance.

**Reply by Iceman248** [reply]
> I have seen this behavior when the reverse proxy is available but not the actual vaultwarden instance.

This would normally be how it would be for me. I can't recall for sure if the reverse proxy was always available when bitwarden wasn't (or at least at the time I tried to use bitwarden), but that would be the case the vast majority of the time.

**Reply by BlackDex** [reply]
> I have seen this behavior when the reverse proxy is available but not the actual vaultwarden instance.

I just tried this, and it doesn't seem to be having any effect for me.
Not on the mobile client or the web-extension, and not even the web-vault it self, they all keep using the cache it looks like.

**Comment by Iceman248** [comment]
I have noticed that if reverse proxy was offline entirely that my bitwarden continued to work in offline mode, so thinking about it, I thought of an idea I am trying to see if it can be done. If I were to modify the CNAME of the domain that is used to access bitwarden, just to break the connection, then bitwarden couldn't access reverse proxy, it would appear down, should still work, but it figures that it seems like A records are the only thing that can be changed automatically. Anyone know how or can think of a different idea to make this happen?

**Reply by Masgalor** [reply]
Maybe it would be a good thing to find out what your reverse-proxy does to trigger this problem.
Since neither @BlackDex  nor I could reproduce this, I tend to blame the proxy configuration for your problems.

Unfortunately I cant help you with your DNS settings as this depends heavily on who manages your domains.
I for example can change and delete any nameserver-entry via API.
I think it would be easier to configure your proxy to send an error code e.g. (403/404/502/503) if you want it to appear offline.

**Reply by Iceman248** [reply]
I use NPM, don't know what would be wrong that would cause an issue like this. It is working as intended based on how I have it set up. It's just that the BW app doesn't like reaching anything at the domain BW is at. It doesn't really have to do with the proxy. I realized it was stopping at the cloud firewall first, so it was never reaching the proxy to have this issue. Even getting rejected at firewall caused the issue. I was able to resolve by making domain no longer valid, so I need to figure out how to automate via API to change domain (ideally CNAME) to gibberish and back to valid when I want it valid. I am using Cloudflare.

**Reply by Iceman248** [reply]
To follow up, I used a shell script to automate the deletion and addition of the DNS record to work around this issue in a satisfactory way. Thanks for the help.

**Reply by rama31244** [reply]
Any chance you can share this shell script? I'm running vaultwarden with a Traefik proxy and notice I lose access to my vault everytime the server goes down as well

**Reply by jjlin** [reply]
There's no need for anything like this. The clients log themselves out if you use the wrong HTTP response, see https://vaultwarden.discourse.group/t/offline-online-access/2298.

**Reply by rama31244** [reply]
Ok thanks. How can I disable the 403 response? I'm using Traefik reverse proxy
